tapgarden/caretaker.go

```diff
@@ -934,12 +965,29 @@ func (b *BatchCaretaker) stateStep(currentState BatchState) (BatchState, error)
 					GenesisProof: mintingProof,
 					Amt:          newAsset.Amount,
 				}
-				_, err = b.cfg.Universe.RegisterIssuance(
-					ctx, uniID, baseKey, mintingLeaf,
+				issuanceBatch = append(
+					issuanceBatch, &universe.IssuanceItem{
+						ID:   uniID,
+						Key:  baseKey,
+						Leaf: mintingLeaf,
+					},
 				)
+
+				// We want a batch to be submitted to the
+				// universe if the batch is full, or we're at
+				// the last asset.
+				if len(issuanceBatch) < batchSize &&
+					idx+1 != len(committedAssets) {
+
+					continue
+				}
+
+				// We have a full batch or are at the latest
+				// asset, so we'll submit the batch to the
+				// universe.
+				err = submitBatch(idx+1, len(committedAssets))
 				if err != nil {
-					return 0, fmt.Errorf("unable to "+
-						"register issuance: %v", err)
+					return 0, err
```

Comment:
I think this also needs to handle indepmotence? May work already, but the scenario is if we have 1k assets to insert, only do 500 of them (the chunk size) then crash. This may return an error that we'll need to be able to gracefully handle. I think it'll just return teh same proof it already has though.

Reply:
Hmm, good question... What do we currently do if the `BatchStateConfirmed` state fails? I assume we retry on next startup and would try to re-insert the proofs, is that what you mean?
I think since we only have upserts in the `universeRegisterIssuance` function, it should already work?

Reply:
Yeah, I think we'll just retry on startup, then the `upserts` and up with us doing nothing. Then we also have the logic that checks to see if an item already exists in the tree first as well. So I think we're fine after all.